mdstat.go

```diff
@@ -0,0 +1,161 @@
+package procfs
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	statuslineRE = regexp.MustCompile(`(\d+) blocks .*\[(\d+)/(\d+)\] \[[U_]+\]`)
+	buildlineRE  = regexp.MustCompile(`\((\d+)/\d+\)`)
+)
+
+// MDStatus holds info parsed from /proc/mdstat.
+type MDStatus struct {
+	// Name of the device.
+	MDName string
+	// activity-state of the device.
+	ActivityState string
```

Comment:
I'd usually vote for shorter names, but if you think `State` would be ambiguous (are there any other md related states?) it's fine to leave it.